lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,173 @@
+        user = self.context.user if self.context.user else "pbench"
+        prefix = self.context.prefix if self.context.prefix else ""
+
+        runs_copied = 0
+        failures = 0
 
-    if runs_copied + failures > 0:
-        logger.debug(
-            "successfully moved %s runs, encountered %s failures", runs_copied, failures
-        )
+        for dirent in config.pbench_run.iterdir():
+            if not self.dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+            result_dir = dirent
+            mrt = MakeResultTb(result_dir, temp_dir, user, prefix, self.config, logger)
+            make_return_val = mrt.check_result_target_dir(result_dir, temp_dir)
+            if make_return_val:
+                sys.exit(1)
+            status, result_tb_name = mrt.make_result_tb()
+
+            if status == 0:
+                if os.path.exists(result_tb_name):
+                    assert (
+                        result_tb_name
+                    ), "Logic bomb!  make_result_tb() always returns a tar ball name"
+                    crt = CopyResultTb(controller, result_tb_name, self.config, logger)
+                    copy_return_val = crt.check_tb_tbmd5()
+                    if copy_return_val:
+                        sys.exit(1)
+                    ret = crt.copy_result_tb(self.context.token)
+                    if ret:
+                        sys.exit(1)
+            else:
+                logger.error("Error occured while creating Result Tarball")
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+                os.remove(f"{result_tb_name}.md5")
+            except OSError:
+                logger.error("rm failed to remove %s and its .md5 file", result_tb_name)
+                failures += 1
+                continue
+
+            try:
+                shutil.rmtree(result_dir)
+            except OSError:
+                logger.error(
+                    "rm failed to remove the %s directory hierarchy", result_dir
+                )
+                failures += 1
+                continue
+
+            runs_copied += 1
+
+        if runs_copied + failures > 0:
+            logger.debug(
+                "successfully moved %s runs, encountered %s failures",
+                runs_copied,
+                failures,
+            )
+            return 1
+
+        return 0
+
+
+@click.command()
+@common_options
+@click.option(
+    "--controller", help="Controller name (should be hostname -f value)",
+)
+@click.option(
+    "--user", help="Pbench user",
+)
+@click.option(
+    "--token",
+    prompt=False,
+    help="pbench server authentication token (will prompt if unspecified)",
```

Comment:
You just turned `prompt=False`, so no, it won't. I'm unsure whether it should, although tokens are relatively long.

Reply:
yeah, I wanted to change it to 'True' . missed that

Reply:
How will it read from a file if `prompt=True`? It should not lead the user to write whole token :P

Reply:
If you include the `required=True` and `prompt=True` options in the command definition, and if the user omits `--token` and does not define the `PBENCH_ACCESS_TOKEN` environment variable, then Click will print a prompt to `stdout` (although, it can be configured to print to `stderr`, instead), and then read from the terminal to get the token value.

I was thinking that, therefore, the user should be able to run the command like the following:
```
$ pbench-move-results --controller my_controller.example.com < my_token_file.txt
```
because Click offers support for mocking input for unit testing (if you look at `test_args_none()` in `lib/pbench/test/unit/agent/task/test_generate_token.py` (e.g., line 121) you can see where the code supplies input for the prompt).

__However__, it turns out that Click is expressly avoiding reading from `stdin` (it reads from the user's _terminal_), so my above approach doesn't work.

This sorta makes sense, as, in the typical case, `stdin` would be input for the program operation, and not responses for command line parsing, so this behavior is probably a Good Thing™...but, it doesn't suit our use case.

Still, the token _is_ required, and so it's probably worth prompting for (although, that's certainly debatable).  I don't expect most humans to supply it (either on the command line or to a prompt), but given the ability to copy-and-paste it, this is not onerous (it's only 139 characters -- it's not like it's 8k...).  I fully expect that it will generally be accessed via the environment variable, but, in the future (beyond the scope of this PR) we might want to add the ability to fetch the token from a file.

Reply:
Ok.. So, believing the ability to copy-paste, I will leave the prompt to be true. I will create an issue regarding this problem for future. Thank you :)

Reply:
Issue #2199.